src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TelemetryBuffer.cs

```diff
@@ -70,6 +72,29 @@ public int Capacity
             }
         }
 
+        /// <summary>
+        /// Gets or sets the maximum number of telemetry items that can be in the backlog to send. Items will be dropped
+        /// once this limit is hit.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">The value is zero or less.</exception>
+        public int MaximumBacklogSize
+        {
+            get
+            {
+                return this.maximumBacklogSize;
+            }
+
+            set
+            {
+                if (value < 1001)
```

Comment:
In the server telemetry channel I believe we need 1000000 as the minimum, not 1001.